### PR TITLE
[TECH SUPPORT] LPS-73752 Asset publisher mail notifications not taking permissions into account

### PR DESCRIPTION
**Pull request by rimisaadouwork**
Hi,

Could you please review this commit.
https://issues.liferay.com/browse/LPS-73752

Thanks!

### CODE REVIEW

**Comment by rimisaadouwork** [comment]
I didn't touched ./modules/apps/web-experience/asset/asset-publisher-web/src/main/resources/META-INF/resources/display/full_content.jsp

File: modules/apps/web-experience/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/messaging/AssetEntriesCheckerUtil.java [hunk 1]
```diff
@@ -286,6 +287,14 @@ private static void _notifySubscribers(
 			}
 
 			for (User user : users) {
+				if (subscriptionSender.getGroupId() > 0) {
+					List<Long> groups = ListUtil.toList(user.getGroupIds());
+
+					if (!groups.contains(subscriptionSender.getGroupId())) {
+						continue;
```
Comment (pavel-savinov) [inline modules/apps/web-experience/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/messaging/AssetEntriesCheckerUtil.java]: Hi @rimisaadouwork,
does this really make sense? Imagine the situation when we have a publicly accessible site with an Asset Publisher and some contents also available for all the users. Any user, even if he is not a member of this site, can subscribe to the AP notifications and should receive them correctly in this case.
And by the way, we have permissions check in the SubscriptionSender:
https://github.com/ealonso/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java#L525

**Comment by rimisaadouwork** [comment]
Hi @pavel-savinov ,
Thank you for your feedback. I will have a second look at it.
Thanks!

**Comment by rimisaadouwork** [comment]
Hi @pavel-savinov,

The problem is when we add an asset to assetpublisher we notify all the users who are subscribed to this portlet. We are not able to check if the the customer has view rights on this asset. 

According to the code we are using subscriptionSender.addRuntimeSubscribers(user.getEmailAddress(), user.getFullName()). We are using this because we don't persist the asset subscription data to the 'subscription' table. We persist only 'com.liferay.portal.kernel.model.PortletPreferences' to the table. So we are not able to use 
permissions check in the SubscriptionSender:
https://github.com/ealonso/liferay-portal/blob/master/portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java#L525
Only if we modify the public accessor of hassPermission(Subscription, user).
In this way we can use it in the 'for (User user : users)' loop.
 
Other solution could be:
We could persist the asset to 'subscription' table but we need to deal with the older subscriptions. We need to persist those somehow to the table. I don't think that this solution will work properly.

If you have already a better solution please let me know.

Thanks!

**Comment by pavel-savinov** [comment]
@rimisaadouwork we always use PortletPreferences to get new asset entries and after that we send the message using SubsxriptionSender. Furthermore, [here](https://github.com/rimisaadouwork/liferay-portal/blob/9f7979503ae2366d443b292238cf2077fdd2cd4a/modules/apps/web-experience/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/messaging/AssetEntriesCheckerUtil.java#L258) we filter our asset entries using PermissionChecker.
It should cover all the cases and if it is not working as expected we could debug it to see what is happening exactly.

**Pull request closed without merging**